nixos/modules/misc/nixpkgs.nix

```diff
@@ -42,12 +44,42 @@ let
     merge = lib.mergeOneOption;
   };
 
-  _pkgs = import ../../.. config.nixpkgs;
+  pkgsType = mkOptionType {
+    name = "nixpkgs";
+    description = "An evaluation of NixPkgs; the top level attribute set of packages";
+    check = builtins.isAttrs;
+  };
+
+  _pkgs =
+    if cfg.pkgs == null
+    then import ../../.. { inherit (cfg) config overlays system; }
```

Comment:
nit: This should be part of the config section, as:

```nix
{
  config = {
    _module.args.pkgs = cfg.pkgs;
    nixpkgs.pkgs = mkDefault (import ../../.. { inherit (cfg) config overlays system; });
  };
}
```

Reply:
I've ditched the `null` and put the default right in the option definition. That seems to just work.